calculator.go

```diff
@@ -1,13 +1,111 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Add takes 2+ numbers and returns the result of adding them together.
+func Add(a, b float64, rest ...float64) float64 {
+	result := a + b
+	for _, item := range rest {
+		result += item
+	}
+
+	return result
+}
+
+// Subtract takes 2+ numbers and returns the result of subtracting the second
+// from the first.  For variadic args, evaluation is (a - b) - c ...
+func Subtract(first float64, second float64, rest ...float64) float64 {
+	result := first - second
+
+	for _, item := range rest {
+		result -= item
+	}
+
+	return result
+
+}
+
+// Multiply multiplies 2+ numbers.  For variadic args, evaluation is (a * b) * c ...
+func Multiply(first float64, second float64, rest ...float64) float64 {
+	result := first * second
+
+	for _, item := range rest {
+		result *= item
+	}
+
+	return result
+
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+// Divide divides 2+ numbers or returns an error if it would divide by 0.
+// For variadic args, evaluation is (a / b) / c ...
+func Divide(first float64, second float64, rest ...float64) (float64, error) {
+	if second == 0 {
+		return 0, errors.New("can't divide by zero")
+	}
+	result := first / second
+
+	for _, item := range rest {
+		if item == 0 {
+			return 0, errors.New("can't divide by zero")
+		}
+		result /= item
+	}
+
+	return result, nil
+}
+
+// Sqrt takes the square root of a nonnegative number or returns an error.
+func Sqrt(a float64) (float64, error) {
+	if a < 0 {
+		return 0, errors.New("can't take the square root of a negative number")
```

Comment:
It would be good to include the actual input value here, wouldn't it? (use `fmt.Errorf`).